public/app/percona/dbaas/components/DBCluster/DBClusterLogsModal/DBClusterLogsModal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Button, Icon, Spinner, useStyles } from '@grafana/ui';
+import { Modal, logger } from '@percona/platform-core';
+import { DBClusterLogsModalProps } from './DBClusterLogsModal.types';
+import { DBClusterService } from '../DBCluster.service';
+import { Messages } from './DBClusterLogsModal.messages';
+import { toggleLogs, transformLogs } from './DBClusterLogsModal.utils';
+import { DBClusterLogs } from '../DBCluster.types';
+import { PodLogs } from './PodLogs/PodLogs';
+import { getStyles } from './DBClusterLogsModal.styles';
+
+export const DBClusterLogsModal: FC<DBClusterLogsModalProps> = ({ dbCluster, isVisible, setVisible }) => {
+  const styles = useStyles(getStyles);
+  const [loading, setLoading] = useState(false);
+  const [logs, setLogs] = useState<DBClusterLogs>({ pods: [] });
+  const [expanded, setExpanded] = useState(false);
+  const getClusterLogs = async () => {
+    if (!dbCluster) {
+      return;
+    }
+
+    try {
+      setLoading(true);
+      setLogs(transformLogs(await DBClusterService.getLogs(dbCluster), logs));
+    } catch (e) {
+      logger.error(e);
+    } finally {
+      setLoading(false);
+    }
+  };
+  const toggleCollapse = () => {
+    setLogs({ pods: toggleLogs(logs.pods, !expanded) });
+    setExpanded(!expanded);
```

Comment:
```suggestion
    setExpanded(expanded => !expanded);
```

Reply:
Not understanding why should I use a function if `setExpanded` expects a boolean.

Reply:
I think it's common practice to do so when the next value depends on the previous, given that state updates are asynchronous:
- https://reactjs.org/docs/hooks-reference.html#functional-updates
- https://stackoverflow.com/questions/61313442/react-hooks-toggle-setstate

Reply:
Agreed, it's better to use the function approach, since it avoids a dependency, but the right suggestion should be:

```suggestion
    setExpanded(currentValue => !currentValue);
```

since it's clearer :)

Reply:
Done guys, thanks for explaining, I forgot about that!!